x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/ReservedSecurityStateHandlerProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security;
+
+import org.elasticsearch.reservedstate.ReservedClusterStateHandler;
+import org.elasticsearch.reservedstate.ReservedClusterStateHandlerProvider;
+
+import java.util.Collection;
+
+/**
+ * Security Provider implementation for the {@link ReservedClusterStateHandlerProvider} service interface
+ */
+public class ReservedSecurityStateHandlerProvider implements ReservedClusterStateHandlerProvider {
+    private final Security plugin;
+
+    public ReservedSecurityStateHandlerProvider() {
+        throw new IllegalStateException("Provider must be constructed using PluginsService");
+    }
```

Comment:
Since security is a module, Java forces us to declare the no-args constructor when SPI is used. It's not ideal, but we should resolve all these when the new plugin infrastructure is finished.